src/OpenLoco/Vehicles/VehicleManager.cpp

```diff
@@ -14,6 +16,37 @@ namespace OpenLoco::VehicleManager
         regs.esi = X86Pointer(&company);
         call(0x004C3A0C, regs);
     }
+
+    // 0x004AF06E
+    void deleteTrain(Vehicles::VehicleHead& head)
+    {
+        Ui::WindowManager::close(Ui::WindowType::vehicle, enumValue(head.id));
+        auto* vehListWnd = Ui::WindowManager::find(Ui::WindowType::vehicleList, enumValue(head.owner));
+        if (vehListWnd != nullptr)
+        {
+            vehListWnd->invalidate();
+            Ui::Windows::VehicleList::removeTrainFromList(vehListWnd, head.id); // 0x004C1D19
+        }
+        // Change to vanilla, update the build window to a valid train
+        auto* vehBuildWnd = Ui::WindowManager::find(Ui::WindowType::buildVehicle, enumValue(head.owner));
+        if (vehBuildWnd != nullptr)
+        {
+            vehBuildWnd->invalidate();
+            Ui::Windows::BuildVehicle::sub_4B92A5(vehBuildWnd);
+        }
+
```

Comment:
Add a part to remove audio refs as well.

Reply:
Was this added? I don't see it

Reply:
No not yet :(